routes/weather.ts

```diff
@@ -262,31 +229,26 @@ export const getWateringData = async function( req: express.Request, res: expres
 		wateringData = await weatherProvider.getWateringData( coordinates );
 	}
 
-	let scale = -1,	rainDelay = -1;
+	const adjustmentMethodResponse: AdjustmentMethodResponse = await adjustmentMethod.calculateWateringScale(
+		adjustmentOptions, wateringData, coordinates, weatherProvider
+	);
 
-	if ( adjustmentMethod === ADJUSTMENT_METHOD.ZIMMERMAN ) {
-		scale = calculateZimmermanWateringScale( adjustmentOptions, wateringData );
+	// If an error that should be sent to the user occurred, send it to them.
+	if ( adjustmentMethodResponse.errorMessage ) {
+		res.send( "Error: " + adjustmentMethodResponse.errorMessage );
+		return;
 	}
 
-	if (wateringData) {
+	// If an error that should be gracefully handled occurred, default to 100% watering.
+	let scale = ( adjustmentMethodResponse.scale !== undefined ) ? adjustmentMethodResponse.scale : 100;
```

Comment:
I am not sure we need this line and 244. I think it would be tidier if the Anjustment Method routines raised exceptions if they cant calculate the correct information and that the endpoint returns control to Firmware.

Reply:
We can discuss this in #54